### PR TITLE
Dungeons enemies thingy not done yet

### PR DESCRIPTION
**Pull request by DarkBrick**
Enemies rework, split them into "zones", each one with a diferent theme
Things to do:

- [x] Tweak stats for later enemies, currently too generic to be worthwhile
- [x] Make ai's
- [ ] Check if its actually balanced

### CODE REVIEW

**Comment by Felix-44** [comment]
gonna do a task list to simplify making the ais
- [ ] slime
- [ ] skeleton
- [ ] rock-elemental
- [ ] goblin
- [ ] bat
- [ ] lost-spirit
- [ ] vine-monster
- [ ] nature-elemental
- [ ] gemstone-golem
- [ ] crystal-elemental
- [ ] great spirit
- [ ] power-ghost
- [ ] void-ghost
- [ ] the-void
- [ ] bone-snake
- [ ] lava-elemental
- [ ] blight-orb
- [ ] radiant-core
- [ ] question-mark
- [ ] error
- [ ] ancient-swordmaster
- [ ] Giygas clone
- [ ] amoled-elemental
- [ ] true-elemental
- [ ] ruined-elemental
- [ ] Calamitas clone

**Comment by DarkBrick** [comment]
cool

**Comment by Felix-44** [comment]
btw, perhaps we could add a check attack that doesnt deal damage, or cost mana, but tells the enemy stats
would still count as your turn so the enemy will attack after you use it

**Comment by DarkBrick** [comment]
good idea

**Comment by DarkBrick** [comment]
inspect attack when

**Comment by DarkBrick** [comment]
yes thats the name of the attack you are proposing

**Comment by Felix-44** [comment]
oke

**Comment by Felix-44** [comment]
"if you use a spell you take some damage"
not sure how I could code that

**Comment by Lemon-Cannon** [comment]
odd

**Comment by DarkBrick** [comment]
lemon going mad with power

**Comment by Lemon-Cannon** [comment]
Made the void the nastiest bitch in the game. He summons groups of enemies that slaughter you, with many different variations, making you easy prey to get overwhelmed. To make sure his minions last, he reduces your mana, making you lose your most valuable asset, attack ground. He also has infinite mana but likes to flex that by wasting turns saying he's at still full mana.

**Comment by Lemon-Cannon** [comment]
100% gonna nerf him later, but for now, that is the boss I'd be expecting at the end of the game

**Comment by DarkBrick** [comment]
neat

**Comment by Felix-44** [comment]
the final boss is the calamitas clone, while we technically have an ai already (the old one), we should probably improve it later

**Comment by Lemon-Cannon** [comment]
gonna nerf void rtn then

**Comment by Lemon-Cannon** [comment]
@Felix-44 check out my blight orb ai

**Comment by Lemon-Cannon** [comment]
It looks correct to me, but its better if you check it first

**Comment by Felix-44** [comment]
@Lemon-Cannon I'll explain some mistakes you did on discord

File: src/Minigames/dungeon.js [hunk 1]
```diff
@@ -404,6 +404,39 @@ Dungeon.thinkers = {
...
+        }
+        return "The sight of their master powers the elementals onward. Their defence sharply rose!"
+    },
+     "ruined-elemental": (DungeonGame, Entity) => {
+        if (GetPercentual() <= 50) {
+            DungeonGame.enemies.push(new Entity(...Dungeon.enemies[61]))
+            return "Its crumbling body splits into 2...It is broken, not dead"
+        }
+        for (let i = 0; i <= DungeonGame.enemies.length; i++) {
+            Entity.fight(DungeonGame.player, Entity.attack)
```
Comment (Felix-44) [inline src/Minigames/dungeon.js:436]: should say to the user the damage they got

**Comment by DarkBrick** [comment]
Ok so we do a little feedbacking
1. Ramping is still too slow, i'll modify enemy stats to fix some of the issues but other changes are needed so:
- [ ] Make it so enemies get harder faster. For example on floor 1 enemies from 1 to 4 can spawn (in this example that would be from green slime to goblin novice) on floor 2 it's 2 to 5 then 3 to 6 and so on
2. Dogecoin rewards from treasures when exploring are low when compared to dogecoins from enemies, should probably either be increased by like 2x or scale exponentionally
- [ ] Therefore magic checkbox for a task to increase treasure drops
3. Make the &dungeon attack menu better looking either via embeds or le funny ```
- [ ] Reformat list of attacks menu
- [ ] Make descriptions a bit shorter because reading is for nerds

**Comment by DarkBrick** [comment]
*proceeds to instantly forget about the 1 thing im suposed to sk*

**Comment by Felix-44** [comment]
this was not actually merged, was it??

**Comment by LJMSU** [comment]
no

**Comment by Felix-44** [comment]
don't we need another pr then

**Comment by LJMSU** [comment]
idk

**Pull request merged**